examples/spawn-at-surface-with-anchors.html

```diff
@@ -0,0 +1,273 @@
+  window.addEventListener('resize', onWindowResize, false);
+  window.addEventListener('touchstart', onClick, false);
+
+  // Kick off the render loop!
+  update();
+}
+
+/**
+ * The render loop, called once per frame. Handles updating
+ * our scene and rendering.
+ */
+function update() {
+  // Clears color from the frame before rendering the camera (arView) or scene.
+  renderer.clearColor();
+
+  // Render the device's camera stream on screen first of all.
+  // It allows to get the right pose synchronized with the right frame.
+  arView.render();
+
+  // Update our camera projection matrix in the event that
+  // the near or far planes have updated
+  camera.updateProjectionMatrix();
+
+  // Update our perspective camera's positioning
+  vrControls.update();
+
+  // Render our three.js virtual scene
+  renderer.clearDepth();
+  renderer.render(scene, camera);
+
+  // Kick off the requestAnimationFrame to call this function
+  // when a new VRDisplay frame is rendered
+  vrDisplay.requestAnimationFrame(update);
+}
+
+/**
+ * On window resize, update the perspective camera's aspect ratio,
+ * and call `updateProjectionMatrix` so that we can get the latest
+ * projection matrix provided from the device
+ */
+function onWindowResize () {
+  camera.aspect = window.innerWidth / window.innerHeight;
+  camera.updateProjectionMatrix();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+}
+
+/**
+ * When clicking on the screen, fire a ray from where the user clicked
+ * on the screen and if a hit is found, place a cube there.
+ */
+function onClick (e) {
+  // Remove cubes with more than one finger.
+  if (e.touches.length > 1 && cubes.length > 0) {
+    scene.remove(cubes[0]);
+    anchorManager.removeAnchorForObject3D(cubes[0]);
+    cubes.splice(0, 1);
+    return;
+  }
+
+  // Create a new cube and place it where the hit test specifies.
+  var cubeClone = cube.clone();
+  // Inspect the event object and generate normalize screen coordinates
+  // (between 0 and 1) for the screen position.
+  var x = e.touches[0].pageX / window.innerWidth;
+  var y = e.touches[0].pageY / window.innerHeight;
+
+  // Send a ray from the point of click to the real world surface
+  // and attempt to find a hit. `hitTest` returns an array of potential
+  // hits.
+  var hits = vrDisplay.hitTest(x, y);
+
+  // If a hit is found, just use the first one
+  if (hits && hits.length) {
+    var hit = hits[0];
+    // Use the `placeObjectAtHit` utility to position
+    // the cube where the hit occurred
+    THREE.ARUtils.placeObjectAtHit(cubeClone,  // The object to place
+                                   hit,   // The VRHit object to move the cube to
+                                   1,     // Easing value from 0 to 1; we want to move
+                                          // the cube directly to the hit position
+                                   true); // Whether or not we also apply orientation
+
+  }
+  anchorManager.addAnchorForObject3D(cubeClone);
```

Comment:
Does this just add a new anchor for <0,0,0> if no hit is found? Maybe it should require a surface? Or just always 'spawn at camera' if we're just testing anchors and not hit testing

Reply:
You are right. I will make it so the cloned cube is only added if a hit happens.